app/models/manageiq/providers/amazon/storage_manager/ebs/cloud_volume.rb

```diff
@@ -100,8 +100,9 @@ def provider_object(connection = nil)
 
   def modify_volume_options(options = {})
     modify_opts = {}
+
     if volume_type != 'standard'
-      modify_opts[:volume_type] = options[:volume_type] if options[:volume_type] && options[:volume_type] != 'standard'
+      modify_opts[:volume_type] = options[:volume_type] if options[:volume_type] && options[:volume_type] != volume_type
       modify_opts[:size]        = Integer(options[:size]) if options[:size] && Integer(options[:size]).gigabytes != size
```

Comment:
There I wrap the exception received from AWS API, yes. Prior to this patch, the API call would not even be made when `standard` volume type is requested. This means that the update would actually success (from MIQ perspective) because AWS API was not made.

Other providers do not actually care what data is being passed to the method - it's more or less verbatim passed to the actual provider API. For example, [here](https://github.com/ManageIQ/manageiq-providers-openstack/blob/master/app/models/manageiq/providers/openstack/cloud_manager/cloud_volume.rb#L36-L39) is the update for OpenStack.